### PR TITLE
[MRG+1] [LinkExtractors] Ignore bogus links (#907)

### PR DESCRIPTION
**Pull request by nyov**
Because issue #907  is still around, this is a rebase of former PR #927 
But I re-wrote the RegexLE modification from that PR.

### CODE REVIEW

File: scrapy/linkextractors/regex.py [hunk 1]
```diff
@@ -14,16 +14,25 @@ def clean_link(link_text):
...
 
+
 class RegexLinkExtractor(SgmlLinkExtractor):
     """High performant link extractor"""
 
     def _extract_links(self, response_text, response_url, response_encoding, base_url=None):
+        def clean_text(text):
+            return replace_escape_chars(remove_tags(text.decode(response_encoding))).strip()
+
+        def clean_url(url):
```
Comment (nramirezuy) [inline scrapy/linkextractors/regex.py]:
```python
        def clean_url(url):
            try:
                return urljoin(base_url, replace_entities(clean_link(url.decode(response_encoding))))
            except ValueError:
                return ''
```

File: scrapy/linkextractors/lxmlhtml.py [hunk 2]
```diff
@@ -49,8 +49,13 @@ def _extract_links(self, selector, response_url, response_encoding, base_url):
         # hacky way to get the underlying lxml parsed document
         for el, attr, attr_val in self._iter_links(selector._root):
             # pseudo lxml.html.HtmlElement.make_links_absolute(base_url)
-            attr_val = urljoin(base_url, attr_val)
-            url = self.process_attr(attr_val)
+            try:
+                attr_val = urljoin(base_url, attr_val)
+            except ValueError:
+                # skipping bogus links
+                url = None
```
Comment (nramirezuy) [inline scrapy/linkextractors/lxmlhtml.py]: just put a `continue` here

Comment (nyov) [inline scrapy/linkextractors/lxmlhtml.py]: oh, absolutely!
Thanks for reviewing

File: scrapy/linkextractors/regex.py (hunk 1 above)
Comment (nyov) [inline scrapy/linkextractors/regex.py]: I just recalled thinking 'there is a reason I'm writing it this way' while rewriting this one.

What do you think about the line throwing anything else than `ValueError`?
Should it return `''` then, or the original url?

Comment (nyov) [inline scrapy/linkextractors/regex.py]: Nevermind, stupid brain got it now :P

**Comment by barraponto** [comment]
We need a warning for skipped links, or else the user might think evil magic is eating some links.

**Comment by nyov** [comment]
And what would be the best way of doing that here? By raising a custom Exception instead of the `ValueError`, using `warnings`, or just importing a logger and logging a debug line?

**Comment by dangra** [comment]
thanks!

**Comment by nyov** [comment]
But what about what @barraponto said, should it log a warning for skipped links?

**Comment by dangra** [comment]
log messages or exceptions are too much, at most it can collect the last N skipped links in a local attribute so user code can check

**Comment by nyov** [comment]
Okay then! I shall leave that for someone else to implement, who might need it :)

**Pull request merged**